zap/src/main/java/org/zaproxy/zap/authentication/AuthenticationMethod.java

```diff
@@ -273,6 +478,81 @@ public void setLoggedOutIndicatorPattern(String loggedOutIndicatorPattern) {
         }
     }
 
+    public AuthCheckingStrategy getAuthCheckingStrategy() {
+        return authCheckingStrategy;
+    }
+
+    public void setAuthCheckingStrategy(AuthCheckingStrategy authCheckingStrategy) {
+        this.authCheckingStrategy = authCheckingStrategy;
+    }
+
+    public String getPollUrl() {
+        return pollUrl;
+    }
+
+    public void setPollUrl(String pollUrl) {
+        this.pollUrl = pollUrl;
+    }
+
+    public String getPollData() {
+        return pollData;
+    }
+
+    public void setPollData(String pollData) {
+        this.pollData = pollData;
+    }
+
+    public String getPollHeaders() {
+        return pollHeaders;
+    }
+
+    public void setPollHeaders(String pollHeaders) {
+        this.pollHeaders = pollHeaders;
+    }
+
+    public int getPollFrequency() {
+        return pollFrequency;
+    }
+
+    public void setPollFrequency(int pollFrequency) {
+        this.pollFrequency = pollFrequency;
+    }
+
+    public AuthPollFrequencyUnits getPollFrequencyUnits() {
+        return pollFrequencyUnits;
+    }
+
+    public void setPollFrequencyUnits(AuthPollFrequencyUnits pollFrequencyUnits) {
+        this.pollFrequencyUnits = pollFrequencyUnits;
+    }
+
+    /**
+     * Gets the last poll result - true means that the user is authnaticated, otherwise false
+     *
+     * @return the last poll result
+     */
+    public Boolean getLastPollResult() {
+        return lastPollResult;
+    }
+
+    /**
+     * Sets the last poll result - this can be used by script or add-ons to change the known logged
+     * in state eg if they have more accurate information
+     *
+     * @param lastPollResult
+     */
+    public void setLastPollResult(Boolean lastPollResult) {
+        this.lastPollResult = lastPollResult;
+    }
+
+    public Date getLastPollTime() {
```

Comment:
Does not seem to be used, really needed?

Reply:
Deliberately exposed in case scripts (or other add-ons) want to know what it is.